sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -1026,27 +1067,37 @@ void gnrc_rpl_send_DAO(gnrc_rpl_instance_t *inst, ipv6_addr_t *destination, uint
         return;
     }
     me = &netif->ipv6.addrs[idx];
-
-    /* add external and RPL FT entries */
-    /* TODO: nib: dropped support for external transit options for now */
-    void *ft_state = NULL;
-    gnrc_ipv6_nib_ft_t fte;
-    while (gnrc_ipv6_nib_ft_iter(NULL, dodag->iface, &ft_state, &fte)) {
-        DEBUG("RPL: Send DAO - building transit option\n");
-
-        if ((pkt = _dao_transit_build(pkt, lifetime, false)) == NULL) {
+    if (mop == GNRC_RPL_MOP_NON_STORING_MODE) {
+        /* Add the prefix to include the parent's global IP address. */
+        memcpy(parent, &dodag->dodag_id, 8);
+        if ((pkt = _dao_transit_build(pkt, lifetime, parent, false)) == NULL) {
             DEBUG("RPL: Send DAO - no space left in packet buffer\n");
             return;
         }
-        if (ipv6_addr_is_global(&fte.dst) &&
-            !ipv6_addr_is_unspecified(&fte.next_hop)) {
-            DEBUG("RPL: Send DAO - building target %s/%d\n",
-                  ipv6_addr_to_str(addr_str, &fte.dst, sizeof(addr_str)), fte.dst_len);
+    }
+    else {
+
+        /* add external and RPL FT entries */
+        /* TODO: nib: dropped support for external transit options for now */
+        void *ft_state = NULL;
+        gnrc_ipv6_nib_ft_t fte;
+        while (gnrc_ipv6_nib_ft_iter(NULL, dodag->iface, &ft_state, &fte)) {
+            DEBUG("RPL: Send DAO - building transit option\n");
 
-            if ((pkt = _dao_target_build(pkt, &fte.dst, fte.dst_len)) == NULL) {
+            if ((pkt = _dao_transit_build(pkt, lifetime, parent, false)) == NULL) {
```

Comment:
I think the `parent` address [isn't needed in stored mode](https://datatracker.ietf.org/doc/html/rfc6550#section-6.7.8):
> A typical storing node will use one Transit Information option with no parent field (...)